ArduCopter/mode.cpp

```diff
@@ -631,6 +631,71 @@ void Mode::land_run_horizontal_control()
         }
     }
 
+    // call attitude controller
+    if (auto_yaw.mode() == AUTO_YAW_HOLD) {
+        // roll & pitch from waypoint controller, yaw rate from pilot
+        attitude_control->input_thrust_vector_rate_heading(loiter_nav->get_thrust_vector(), target_yaw_rate);
+    } else {
+        // roll, pitch from waypoint controller, yaw heading from auto_heading()
+        attitude_control->input_thrust_vector_heading(loiter_nav->get_thrust_vector(), auto_yaw.yaw());
+    }
+}
+
+#if PRECISION_LANDING == ENABLED
+// Go towards a position commanded by prec land state machine
+// Also let user take over the controls if they want to
+void Mode::land_retry_position(const Vector3f &retry_loc)
+{
+    float target_roll = 0.0f;
+    float target_pitch = 0.0f;
+    float target_yaw_rate = 0;
+    if (!copter.failsafe.radio) {
+        if ((g.throttle_behavior & THR_BEHAVE_HIGH_THROTTLE_CANCELS_LAND) != 0 && copter.rc_throttle_control_in_filter.get() > LAND_CANCEL_TRIGGER_THR){
+            AP::logger().Write_Event(LogEvent::LAND_CANCELLED_BY_PILOT);
+            // exit land if throttle is high
+            if (!set_mode(Mode::Number::LOITER, ModeReason::THROTTLE_LAND_ESCAPE)) {
+                set_mode(Mode::Number::ALT_HOLD, ModeReason::THROTTLE_LAND_ESCAPE);
+            }
+        }
+
+        if (g.land_repositioning) {
+            // apply SIMPLE mode transform to pilot inputs
+            update_simple_mode();
+
+            // convert pilot input to lean angles
+            get_pilot_desired_lean_angles(target_roll, target_pitch, loiter_nav->get_angle_max_cd(), attitude_control->get_althold_lean_angle_max());
+
+            // record if pilot has overridden roll or pitch
+            if (!is_zero(target_roll) || !is_zero(target_pitch)) {
+                if (!copter.ap.land_repo_active) {
+                    AP::logger().Write_Event(LogEvent::LAND_REPO_ACTIVE);
+                }
+                copter.ap.land_repo_active = true;
+            }
+        }
+
+        // get pilot's desired yaw rate
+        target_yaw_rate = get_pilot_desired_yaw_rate(channel_yaw->get_control_in());
+        if (!is_zero(target_yaw_rate)) {
+            auto_yaw.set_mode(AUTO_YAW_HOLD);
+        }
+    }
+
+    if (!copter.ap.land_repo_active) {
+        pos_control->set_pos_target_xy_cm(retry_loc.x, retry_loc.y);
```

Comment:
this is wrong as a soon as the use will release the land repositioning, the drone will go back to the retry_loc ...

Reply:
Yes, you are absolutely right, I didn't think about this. I am not confident about how I have been using the position controller. It will need some changes and advice from others

Reply:
IMHO, don't give the hand to the user. Like on normal precland, if the user want to use the joystick, it switch to normal LAND, mixing auto and manual feature is rarely a good thing

Reply:
100% agreed, removed the manual takeovers inside autonomous retries. The new logic is, if during a retry, user gives rc command, retry is cancelled and it proceeds to failsafe measure.